spec/services/discord/clear_roles_service_spec.rb

```diff
@@ -0,0 +1,50 @@
+require 'rails_helper'
+
+describe Discord::ClearRolesService do
+  subject { described_class }
+
+  let(:discord_configuration) do
+    { discord: { bot_token: 'bot_token', server_id: 'server_id' } }
+  end
+
+  let(:school) { create :school, configuration: discord_configuration }
+
+  let(:user) do
+    create :user,
+           school: school,
+           discord_user_id: Faker::Number.number(digits: 18)
+  end
+
+  describe '#execute' do
+    context 'when a user has a discord user id' do
+      it 'clears the roles' do
+        expect(Discordrb::API::Server).to receive(:update_member).with(
+          "Bot #{discord_configuration[:discord][:bot_token]}",
+          discord_configuration[:discord][:server_id],
+          user.discord_user_id,
+          roles: []
+        )
+
+        subject.new(user, user.discord_user_id).execute
+      end
+    end
+
+    context 'when a user does not have a discord user id' do
+      it 'does not clear the roles' do
+        expect(Discordrb::API::Server).not_to receive(:update_member)
+
+        subject.new(user, nil).execute
+      end
+    end
+
+    # when configuration is not present, the service should not clear roles
+    context 'when configuration is not present' do
+      before { school.configuration = {} }
```

Comment:
Something weird is going on here. This record hasn't been saved. The same issue is present in other spec files.